### ISSUE TITLE
Could we add `'∠'` character as a power operator

### ISSUE THREAD

**Issue by i-apellaniz** [JuliaLang/julia#41397]
https://github.com/JuliaLang/julia/blob/74fab49ffba4913fd93edb742ed53c5728bc4c42/src/julia-parser.scm#L30

Could we add `'∠'` as a power operator so that one can define some function to create complex numbers, for example,
```julia
julia> 1 * ℯ^(im*π) == 1 ∠ π
true
```
I think here is room for discussion; and this other issue is also related somehow https://github.com/JuliaLang/julia/issues/35538

**Comment by JeffBezanson** [comment]
This seems to be an identifier character currently, so technically breaking.

**Comment by stevengj** [comment]
Yes, I don't think we can change this in Julia 1.x.

**Comment by ess3sq** [comment]
> Yes, I don't think we can change this in Julia 1.x.

Could we perhaps consider another symbol?

**Comment by stevengj** [comment]
Is there any other commonly used symbol for this?

`r * cis(φ)` seems pretty straightforward already for polar complex numbers…

**Comment by ess3sq** [comment]
> `r * cis(φ)` seems pretty straightforward already for polar complex numbers…

I'll try to hijack this issue for a short while in order to ask if you'd consider a shorthand `rcis` function for lazy People like me which basically goes like

```julia
rcis(r, phi) = r * cis(phi)
```

unless there is already a constructor for `Complex{T}` which takes polar coordinates?

**Comment by stevengj** [comment]
`rcis(r,phi)` takes *more* characters than `r*cis(phi)`… or *one* fewer if `r` is an expression and you need to write `(r)*cis(phi)`.   It really doesn't seem worth it to add a function to the standard library for this.

**Comment by ess3sq** [comment]
> `rcis(r,phi)` takes _more_ characters than `r*cis(phi)`… or _one_ fewer if `r` is an expression and you need to write `(r)*cis(phi)`. It really doesn't seem worth it to add a function to the standard library for this.

That's a fair enough point, but I would still like to make the claim that having such a function might _complete_ the set of functions that make up the Julia API for complex numbers because afaik there is no function that constructs a `Complex{T}` from polar coords.

Additionally, this could be handy if one, say, was making some kind of calculation that yielded radii and angles, and wants to calculate the actual complex number, like so:

```julia
phi = calc_angles()
r = calc_radii()
z = map(rcis, r, phi)
```

Of course one can do it very easily using dot operators

```julia
z = r .* cis.(phi)
```

though this is aesthetically less pleasing (granted, that is subjective) and perhaps less efficient (broadcasting twice instead of once... unless the Julia compiler can optimise them together...?).

**Comment by giordano** [comment]
> unless the Julia compiler can optimise them together...?

Yes, nested broadcasts are fused together: [More Dots: Syntactic Loop Fusion in Julia](https://julialang.org/blog/2017/01/moredots/)

**Comment by ess3sq** [comment]
I guess this issue has not gotten much attention lately.

> > unless the Julia compiler can optimise them together...?
> 
> Yes, nested broadcasts are fused together: [More Dots: Syntactic Loop Fusion in Julia](https://julialang.org/blog/2017/01/moredots/)

This makes sense so I cannot argue anything about performance, I suppose, but I believe this still stands:
> Of course one can define the rcis function or pass a lambda to map, but this looks common enough that it may warrant some consideration to be standardised and added to Base.

In summary, adding `rcis(r, phi)` would offer not drawbacks and would make `map`'ing easier.
